src/core/coap/coap.cpp

```diff
@@ -309,7 +309,7 @@ otError CoapBase::SendHeaderResponse(Message::Code aCode, const Message &aReques
 
     default:
         ExitNow(error = OT_ERROR_INVALID_ARGS);
-        break;
+        OT_UNREACHABLE_CODE(break;)
```

Comment:
I wonder if it is necessary to keep the `break` after `ExitNow()` statements.

Reply:
I had figured it was a code style requirement in any switch statement to avoid possible fall throughs.

Reply:
Yeah, I think I remember needing to include these to avoid static code analysis warnings as well.